### PR TITLE
make color class serializable

### LINKED ISSUE #285: Make Color class serializable
Color class should be serializable.
Currently, the Sprite class must serialize the color itself instead of delegating the process to the Color object.
An option to choose how you want to serialize the color (As RGB, HSL, Name, Hex) would be nice.

### PR DESCRIPTION
**Pull request by PierrickLP**
Color class is now directly serializable.
It is possible to choose which format to dump to between those four formats : Hex, Rgba, ColorName, Hsv
`color:dump(obe.Graphics.ColorType.Hex)`

### CODE REVIEW

**Review by Sygmei — COMMENTED** [review]
Excellent work :) Could you just add a few unit tests ?

File: src/Core/Graphics/Sprite.cpp [hunk 1]
```diff
@@ -549,33 +547,8 @@ namespace obe::Graphics
...
-            else if (color.is<vili::string>())
-            {
-                spriteColor.fromString(color);
-            }
-            else
-            {
-                throw Exceptions::InvalidSpriteColorType(
-                    vili::to_string(color.type()), color.dump(), EXC_INFO);
-            }
+            Color spriteColor;
```
Comment (Sygmei) [inline src/Core/Graphics/Sprite.cpp:550]: Why not
```cpp
m_color.load(data.at("color"));
```
?

File: src/Core/Graphics/Color.cpp [hunk 2]
```diff
@@ -399,6 +469,91 @@ namespace obe::Graphics
...
+    {
+        std::stringstream ss;
+        ss << "#";
+        ss << std::hex << (int(r + 0.5) << 16 | int(g + 0.5) << 8 | int(b + 0.5));
+        return ss.str();
+    }
+
+    std::string Color::toString() const
+    {
+        std::string res;
```
Comment (Sygmei) [inline src/Core/Graphics/Color.cpp:551]: Maybe
```cpp
const std::string name = toNamed();
if (res.empty())
{
    return toHex();
}
return name;
```

File: src/Core/Graphics/Sprite.cpp (hunk 1 above)
Comment (PierrickLP) [reply src/Core/Graphics/Sprite.cpp:550]: Sprite doesn't have an m_color member

Comment (Sygmei) [reply src/Core/Graphics/Sprite.cpp:550]: Ah, right, it uses sf::Sprite

**Pull request merged**